src/components/ui/Table/Table.tsx

```diff
@@ -0,0 +1,185 @@
+/* eslint-disable react/jsx-key */
+import { memo, useEffect } from 'react';
+import type {ReactNode } from 'react'
+import { useTable, usePagination } from 'react-table';
+import type { TableHeaderProps, HeaderPropGetter, Column } from 'react-table'
+
+import classnames from 'classnames';
+
+import { DEFAULT_PAGE_SIZE } from '../../../constants';
+
+import Pagination from './components/Pagination';
+import TokenPagination from './components/TokenPagination';
+import type { PaginationOptions, TokenPaginationOptions } from './constants';
+import type { PaginationControl } from './components/Pagination';
+
+import styles from './Table.module.scss';
+
+// REF https://github.com/TanStack/react-table/discussions/2664
+type ColumnOptions = {
+  accessor: string | (() => void),
+  Header?: string | (() => void),
+  isNumeric?: boolean,
+  isHidden?: boolean,
+  lightWeight?: boolean
+  boldHeader?: boolean
+  bold?: boolean
+  className?: string,
+  customWidth?: number,
+}
+
+type ColumnData = {
+  [key: string]: string
+} & Omit<ColumnOptions, "accessor" | "Header">
```

Comment:
Do we need `Omit` part here? Isn't it enough only `[key: string]: string` since data is just an array like `[{name: 'John'}]`. There won't be any additional properties from `ColumnOptions`